lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -558,26 +613,17 @@ def set_zsort(self, zsort):
         self.stale = True
 
     def get_vector(self, segments3d):
-        """Optimize points for projection."""
-        si = 0
-        ei = 0
-        segis = []
-        points = []
-        for p in segments3d:
-            points.extend(p)
-            ei = si + len(p)
-            segis.append((si, ei))
-            si = ei
-
-        if len(segments3d):
-            xs, ys, zs = zip(*points)
-        else :
-            # We need this so that we can skip the bad unpacking from zip()
-            xs, ys, zs = [], [], []
-
-        ones = np.ones(len(xs))
-        self._vec = np.array([xs, ys, zs, ones])
-        self._segis = segis
+        """Optimize points for projection"""
+
+        self._seg_sizes = np.array([len(c) for c in segments3d])
+        self._vec = []
+        if len(segments3d) > 0:
+            # Store the points in a single array for easier projection
+            n_segments = np.sum(self._seg_sizes)
+            # Put all segments in a big array
+            _vec = np.vstack(segments3d)
+            # Add a fourth dimension
+            self._vec = np.hstack([_vec, np.ones((n_segments, 1))]).T
```

Comment:
More efficient as
```
self._vec = np.empty((n_segments, 4))
self._vec[:3,:] = segments3d
self._vec[3,:] = 1
```

which also eliminates the need for the special case when `len(segments3d) == 0`